src/util/scroll.js

```diff
@@ -3,63 +3,64 @@ import { changeCenter } from './path.js';
 
 export const actions = {
     exit: (response) => {
-        response.element.classList.remove('active');
+        const { element } = response;
+        element.classList.remove('active');
     },
     enter: (response, config, map) => {
-        const { element } = response;
-        element.classList.add('active');    
-        
-        const chapter = config.chapters.find(
-            (chap) => chap.id === response.index
-        );
-        const speciesList = element.querySelectorAll('.icon');;
-    
-        Array.from(speciesList).forEach( (item, idx) => {
+        const { index, element } = response;
+        element.classList.add('active');
+
+        [...element.querySelectorAll('.icon')].forEach((item, idx) => {
             item.style.animationTimingFunction = "ease-in-out";
             item.style.animationDelay = `${idx * 0.1}s`;
         });
 
         if (element.className.includes('chapter')) {
-            map.getSource('pointSource').setData(point(chapter.location.center));
-            map.getSource('pointSourceBg').setData(point(chapter.location.center));
+            const chapter = config.chapters.find((chap) => chap.id === index);
+            const center = point(chapter.location.center)
+            map.getSource('pointSource').setData(center);
+            map.getSource('pointSourceBg').setData(center);
         }
-   
     },
     progress: (response, config, map, data) => {
-
         const { index, progress, element } = response;
-        let stepProgress;
-        let movingLine;
-    
-        const chapter = config.chapters.find(
-            (chap) => chap.id === response.index
-        );
-    
-    
+        const chapter = config.chapters.find(chapter => chapter.id === index);
+
+        const atBottom = element.className.includes('footer')
+            || document.body.offsetHeight <= window.innerHeight + window.scrollY;
+
         if (element.className.includes('header')) {
-            response.element.style.opacity = Math.round(1 - response.progress);
+            element.style.opacity = Math.round(1 - progress);
+        }
+
+        const setCenter = center => {
+            if (!config.followPoint || !center || atBottom) return;
+
+            const { lng, lat } = map.getCenter()
+            const previous = [lng, lat]
+            const round = x => Math.round(x * 1e3)
+            const isSame = String(previous.map(round)) === String(center.map(round))
+            if (isSame) return;
+
+            map.easeTo({ center, zoom: 13, bearing: chapter.location.bearing, pitch: 65 });
+        }
+
+        const finishedSteps = Math.max(0, index - 1)
```

Comment:
```suggestion
        const currentIndex = Math.max(0, index - 1)
```

maybe this should be called something like `currentIndex` since it refers to the index and not the progress steps

Reply:
I was thinking of this variable as "the number of previous slides / path segments that are completed to the end and need to be copied in their entirety", as opposed to the current one, which only needs to be copied partially. Thinking about numbers of items is generally more natural for me than indices) The word `index` also sounds very generic to my ear (unless it's used in the database sense), so I try to avoid it and come up with something more descriptive.

I don't mind changing it if you prefer another approach, of course)